tests/integration_tests/test_guard.py

```diff
@@ -903,3 +904,20 @@ def test_guard_with_top_level_list_return_type(mocker, rail, prompt):
         {"name": "banana", "price": 0.5},
         {"name": "orange", "price": 1.5},
     ]
+
+
+def test_pydantic_with_lite_llm(mocker):
+    """Test lite llm JSON generation with message history re-asking."""
+    mocker.patch(
+        "guardrails.llm_providers.LiteLLMCallable",
+        new=MockLiteLLMChatCallable,
+    )
```

Comment:
Let's not perpetuate this pattern and instead write the mocks directly in the test using `mock.side_effect` if we need to declare multiple returns.

Reply:
to be painfully clear you mean like 
```
class MyTestMockLiteLLMChatCallable(OpenAIChatCallable):
    ... mock some things

mocker.patch(
    "guardrails.llm_providers.LiteLLMCallable",
    new=MockLiteLLMChatCallable,
)
```
?

Reply:
nvm you mean
```
    mock_invoke_llm = mocker.patch(
        "guardrails.llm_providers.OpenAIChatCallable._invoke_llm"
    )
    mock_invoke_llm.side_effect = [
        LLMResponse(
            output=pydantic.MSG_HISTORY_LLM_OUTPUT_INCORRECT,
            prompt_token_count=123,
            response_token_count=1234,
        ),
        LLMResponse(
            output=pydantic.MSG_HISTORY_LLM_OUTPUT_CORRECT,
            prompt_token_count=123,
            response_token_count=1234,
        ),
    ]
```

Reply:
Yes, sorry didn't see your first message